### DISCUSSION TITLE
QNap Lessons learned / Semi Guide

### DISCUSSION THREAD

**Discussion by fanno** [immich-app/immich#13430] [Community Guides]
QNap TS-4733A 8 GB ram (upgrated to 32 GB), QUTS 5 hero
System using around 9GB running "idle" so is the default 8 GB is propelfly not enough. i do have other services running
NOTE: it uses a lot more when running new images and so on. My best guess would be 16 GB minimum

Requirements
Internet access on the Qnap (while installing at least)
Container Station (mine was 3.0.8.981)
Portainer (mine was ce 2.21.3)

Initially i tried to install immich directly in to Container Station, however when immich had a update i had problems updating.
I was able to pull a new image but was not able to then figure out how to update it. (this could be user error on my part)

I ended up installing a portainer docker container. (if you use other dockers in qnap, with my limited experiance would be that the container station is not up to par and i would use portainer for everything going forward.) (it dose appear they can coexist)

Container Station > Application > Create
Application name = "portainer" or what ever makes sense to you

YAML
To be fair and up front i am new to docker so i am not familiar with all the variables below

Ports:
I change the port to another port because i had a conflict. on that port. (another service i have on the NAS)
volumes:
I changed the /data from ./data to /share/Container/portainer-ce/data
Container is my shared folder with all my different current and dockers.
environment:
I changed the TZ to one near you see: https://confluence.atlassian.com/kb/how-to-set-the-timezone-for-docker-container-976780914.html

```
version: '3'

services:
  portainer:
    image: portainer/portainer-ce:latest
    container_name: portainer
    restart: always
    security_opt:
      - no-new-privileges:true
    ports:
      - 8050:9000
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - /share/Container/portainer-ce/data:/data:rw
    environment:
      TZ: Europe/Berlin
```

http://ip.address.of.nas:8050/

After this follow the guide: (read the added stuff here below first)
https://immich.app/docs/install/portainer/

QNAP changes i founed required:
You must take care where the data is stored. Make sure chose a dedicated location in a shared folder.  i created a shared called immich
so it would be /share/immich/....
I accidentally had done a typo so it tried to put the whole immich software of the 400 MB of space in the qnap internal storage. (this is a bad thing)
So make sure the paths below is correct via ENV or directly in the defenition

Not it is the same thing for immich-server / immich-machine-learning / database

immich-server:
```
    volumes:
      # Do not edit the next line. If you want to change the media storage location on your system, edit the value of UPLOAD_LOCATION in the stack.env file
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
```

immich-machine-learning:
```
    volumes:
      - ${MODEL_CACHE_LOCATION}:/cache
```

database:
```
    volumes:
      # Do not edit the next line. If you want to change the database storage location on your system, edit the value of DB_DATA_LOCATION in the stack.env file
      - ${DB_DATA_LOCATION}:/var/lib/postgresql/data
```

env:
```
UPLOAD_LOCATION=/share/immich/upload
MODEL_CACHE_LOCATION=/share/immich/model-cache
DB_DATA_LOCATION=/share/immich/pgdata
```

Other notes:
My QNAP is blocked via firewall to the internet.
My firewall was wrong and there was still internet (or check happen while firewall was down while i change something)

To check if the conection is blocked.
portainer:8050 >  local > containers >immich_server > console > connect
then type: apt update
```
0% [Connecting to deb.debian.org (151.101.194.132)]  
...
W: Some index files failed to download. They have been ignored, or old ones used instead.
```
This indicate that there are no internet access if you have similar setup
Immich "team" hinted  when i ask of a way to test this that maybe ping / wget  / curl could be added in feature for a better way to test

Zeus from discoard also suggested this: (i have not yet tried this part)
```
download the file and make sure to chmod 755 so it can be executed
add to the docker volumes section, such as - /path/to/curl:/curl:ro
open a shell with docker exec -it immich_server bash
run the command: /curl https://google.com

```

**Comment by nivek1385** [comment]
This was quite helpful and reduced my headaches setting up Immich.

**Reply by fanno** [reply]
Is any changes needed?

**Reply by nivek1385** [reply]
Just make sure to be careful with the storage location paths in both the portainer and immich setups...

**Reply by deejaywoody** [reply]
Just finished setting-up on a QNAP TS-451+ and this was crazy helpful. Got it going on the first try 🙌🏻

**Reply by Kammandur** [reply]
On the storage paths, do I need to add the volumes in the setting of portainer or will the .yml file take care of the mapping from portainer to my uploads location? When I check portainer in the volumes section there are no volumes map to my uploads location. Just the folder where portainer is installed
